### ISSUE TITLE
'uncommitted' level lints

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#293]
Errors that only appear on uncommitted code. Errors won't show for lines that are committed.

I suppose any rule could be configured to this level, but one rule that would be default is "type could be wider/narrower"
```py
def f(a: str):  # error: type could be wider: "object"
    print(a) 
def f(l: list[object]):  # error: type could be narrower: "Sequence[object]"
    print(l[0])
```

**Comment by DetachHead** [comment]
related (but different use case): #2

**Comment by DetachHead** [comment]
yeah i think this is basically just the same as baseline. perhaps the docs can be updated to mention this use case

**Comment by DetachHead** [comment]
i think the "hint" [diagnostic category](https://docs.basedpyright.com/latest/configuration/config-files/#diagnostic-categories) covers this actually
